polar2grid/writers/hdf5.py

```diff
@@ -0,0 +1,302 @@
+from __future__ import annotations
+from typing import TextIO
+
+import os
+import sys
+import logging
+
+import pkg_resources
+import satpy
+import h5py
+import numpy as np
+import xarray as xr
+import dask.array as da
+
+from itertools import groupby
+from datetime import datetime as datetime
+from satpy.dataset import DataID
+from satpy.writers import ImageWriter
+from satpy.writers import compute_writer_results
+
+from trollsift import Parser
+
+from pyresample.geometry import SwathDefinition
+from polar2grid.utils.legacy_compat import convert_p2g_pattern_to_satpy
+from polar2grid.writers.geotiff import NUMPY_DTYPE_STRS, NumpyDtypeList, str_to_dtype, int_or_float
+
+from polar2grid.core.script_utils import setup_logging, rename_log_file, create_exc_handler
+
+USE_POLAR2GRID_DEFAULTS = bool(int(os.environ.setdefault("USE_POLAR2GRID_DEFAULTS", "1")))
+
+LOG = logging.getLogger(__name__)
+
+# reader_name -> filename
+DEFAULT_OUTPUT_FILENAMES = {None: "{platform_name}_{sensor}_{start_time:%Y%m%d_%H%M%S}.h5"}
+
+
+def dist_is_editable(dist) -> bool:
+    """Is distribution an editable install?"""
+    for path_item in sys.path:
+        egg_link = os.path.join(path_item, dist.project_name + ".egg-link")
+        if os.path.isfile(egg_link):
+            return True
+    return False
+
+
+dist = pkg_resources.get_distribution("polar2grid")
+if dist_is_editable(dist):
+    p2g_etc = os.path.join(dist.module_path, "etc")
+else:
+    p2g_etc = os.path.join(sys.prefix, "etc", "polar2grid")
+
+config_path = satpy.config.get("config_path")
+if p2g_etc not in config_path:
+    satpy.config.set(config_path=config_path + [p2g_etc])
+
+
+def all_equal(iterable: list[str]) -> bool:
+    "Returns True if all the elements are equal to each other"
+    g = groupby(iterable)
+    return next(g, True) and not next(g, False)
+
+
+class FakeHDF5:
+    def __init__(self, fh : TextIO[IO[AnyStr]], output_filename: str, var_name: str, compression: bool):
+        self.fh = fh
+        self.output_filename = output_filename
+        self.var_name = var_name
+        self.compression = compression
+
+    def check_variable(self, data: da.array, append: bool):
+        if append:
+            if self.var_name in self.fh:
+                LOG.warning("Product %s already exists in hdf5 group, will delete existing dataset", self.var_name)
+                del self.fh[self.var_name]
+        self.fh.create_dataset(self.var_name, shape=data.shape, dtype=data.dtype, compression=self.compression)
+
+    def __setitem__(self, write_slice, data):
+        self.fh[self.var_name][write_slice] = data
+
+
+class hdf5writer(ImageWriter):
+    """Writer for hdf5 files."""
+
+    def __init__(self, dtype=None, **kwargs):
+        """Init the writer."""
+        super(hdf5writer, self).__init__(**kwargs)
+        self.dtype = self.info.get("dtype") if dtype is None else dtype
```

Comment:
Just to avoid unnecessary complexity in Satpy in the future, can you make `dtype` an argument on `save_datasets`?